pkg/signup/service/signup_service.go

```diff
@@ -83,11 +94,10 @@ func (s *ServiceImpl) newUserSignup(ctx *gin.Context) (*toolchainv1alpha1.UserSi
 		}
 	}
 
-	cfg := configuration.GetRegistrationServiceConfig()
-
-	verificationRequired := cfg.Verification().Enabled()
+	verificationRequired, captchaScore := IsPhoneVerificationRequired(s.CaptchaChecker, ctx)
 
 	// Check if the user's email address is in the list of domains excluded for phone verification
+	cfg := configuration.GetRegistrationServiceConfig()
 	emailHost := extractEmailHost(userEmail)
```

Comment:
Do we have to do the captcha call if the email domain is excluded from verification?
What do you think about setting the verificatioRequired to false if the email domain is excluded and do not call `IsPhoneVerificationRequired()` at all in this case? The captcha annotation wouldn't be set in this case but does it matter if the verification is not required anyway?

But if it's tricky to organize the code this way or can potentially break anything I'm fine with leaving it as is. It's a minor optimization anyway.